File: Problem-2/PalindromeString.java
// WAP to find whether string is palindrome or not. 

import java.util.*;

public class PalindromeString{
    public static void main(String[] args){

        Scanner sc = new Scanner(System.in);

        System.out.println("Enter a string : ");
        String str = sc.next();

        String reverse = "";

        for(int i = str.length() - 1; i >= 0; i--)
            reverse += str.charAt(i);

        System.out.println(str.equals(reverse) ? "Palindrome !!" : "Not Palindrome !!");
    }
}

File: Problem-2/PalindromeString.c
// WAP to find whether string is palindrome or not. 

#include <stdio.h>
#include <string.h>

void main() {
    char str[100];
    int len, flag = 0;

    printf("Enter a string: ");
    scanf("%s", str);

    len = strlen(str);

    for(int i = 0; i < len / 2; i++) {
        if(str[i] != str[len - i - 1]) {
            flag = 1;
            break;
        }
    }

    if(flag == 0){
        printf("Palindrome !!\n");
    }
    else{
        printf("Not Palindrome !!\n");
    }
}